backend/notifications/notificationGenerator.js

```diff
@@ -0,0 +1,115 @@
+import { prisma } from "../utils/helpers.js";
+
+const generateAllNotifications = async (userId) => {
+  const user = await prisma.user.findUnique({
+    where: { id: userId },
+    include: {
+      userInterests: { include: { interests: true } },
+      workoutSessions: { orderBy: { date: "desc" }, take: 5 },
+      followers: {
+        orderBy: { createdAt: "desc" },
+        include: { follower: true },
+        take: 1,
+      },
+    },
+  });
+
+  if (!user || !user.userInterests) return [];
+
+  const daysSinceLastWorkout = Math.floor(
```

Comment:
since we're repeating this calculation for `daysSinceLastWorkout`, do we want to put this in a separate helper function or util file?

Reply:
Thanks for catching that. I will make it one function